terraform-db-roles/dynamodb.tf

```diff
@@ -0,0 +1,24 @@
+data "aws_iam_policy_document" "ehr-transfer-tracker-db-access" {
+  statement {
+    actions = [
+      "dynamodb:GetItem",
+      "dynamodb:PutItem",
+      "dynamodb:UpdateItem",
+      "dynamodb:Query"
+    ]
+    resources = [
+      "arn:aws:dynamodb:${var.region}:${data.aws_caller_identity.current.account_id}:table/${var.environment}-ehr-transfer-tracker"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "ehr-transfer-tracker-db-access" {
+  name   = "${var.environment}-${var.component_name}-transfer-tracker-db-access"
+  policy = data.aws_iam_policy_document.ehr-transfer-tracker-db-access.json
+}
+
+# Grant ECS Task permissions to access the transfer tracker db
+resource "aws_iam_role_policy_attachment" "dynamodb_application_user_policy_attach" {
+  role       = "${var.environment}-${var.component_name}-EcsTaskRole"
+  policy_arn = aws_iam_policy.ehr-transfer-tracker-db-access.arn
+}
```

Comment:
I notice sometimes you've named the resource identifiers with hyphens and sometimes with underscores. Although there's no strict guideline on how to name identifiers, generally underscores are preferred, which is how we've been naming new Terraform code we've been writing.

```suggestion
data "aws_iam_policy_document" "ehr_transfer_tracker_db_access" {
  statement {
    actions = [
      "dynamodb:GetItem",
      "dynamodb:PutItem",
      "dynamodb:UpdateItem",
      "dynamodb:Query"
    ]
    resources = [
      "arn:aws:dynamodb:${var.region}:${data.aws_caller_identity.current.account_id}:table/${var.environment}-ehr-transfer-tracker"
    ]
  }
}

resource "aws_iam_policy" "ehr_transfer_tracker_db_access" {
  name   = "${var.environment}-${var.component_name}-transfer-tracker-db-access"
  policy = data.aws_iam_policy_document.ehr-transfer-tracker-db-access.json
}
```

Reply:
Thanks for spotting this! Just changed the hyphens to underscore